strawberryfields/decompositions.py

```diff
@@ -233,7 +233,10 @@ def nullTi(m, n, U):
     if nmax != mmax:
         raise ValueError("U must be a square matrix")
 
-    if U[m, n+1] == 0:
+    if (U[m, n+1] == 0 and U[m, n] == 0):
```

Comment:
```suggestion
    if (U[m, n] == 0):
```
The proposed handling of this situation correct even if `U[m, n+1] != 0` and makes the code easier to read and saves a couple of function calls.